src/approaches/nsrt_rl_approach.py

```diff
@@ -49,33 +65,215 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
+                # not supposed to change, do change. This can happen in two
+                # ways: 1) if any of the features of the option's objects that
+                # aren't supposed to change, change, and 2) if any feature of
+                # any other objects in the state, change.
+                prev_state_unchanging_feat = _create_absolute_option_param(
+                    curr_states[-2],
+                    var_to_unchanging_feat_ind,
+                    parent._changing_var_order,  # pylint: disable=protected-access
+                    var_to_obj,
+                )  # concatenated unchanging features of the option's objects
+                curr_state_unchanging_feat = _create_absolute_option_param(
+                    curr_states[-1],
+                    var_to_unchanging_feat_ind,
+                    parent._changing_var_order,  # pylint: disable=protected-access
+                    var_to_obj,
+                )  # concatenated unchanging features of the option's objects
+                other_objects = [
+                    o for o in s.data.keys() if o not in curr_option.objects
+                ]
+                # The first term checks the current option's objects, and the
+                # second term checks all other objects.
+                option_objects_unchanged = np.allclose(
+                    prev_state_unchanging_feat - curr_state_unchanging_feat,
+                    0,
+                    atol=1e-5)
+                other_objects_unchanged = np.allclose(
+                    curr_states[-2].vec(other_objects),
+                    curr_states[-1].vec(other_objects),
+                    atol=1e-5)
+                if not option_objects_unchanged or not other_objects_unchanged:
+                    reward += self._neg_reward * 10
+
+                # Add pos_reward if we got within epsilon of the option's
+                # subgoal, otherwise we add neg_reward.
+                # Ignore the optimization that checks for repeated states, which
+                # messes with the terminal state check.
+                _ = curr_option.memory.pop("last_state", None)
+                if curr_option.terminal(s):
+                    # Check if we reached our subgoal within a tolerance.
+                    if np.allclose(relative_param,
+                                   0,
+                                   atol=self._reward_epsilon):
+                        curr_rewards.append(reward + self._pos_reward)
+                    else:
+                        curr_rewards.append(reward + self._neg_reward)
+
+                    # Store transition data.
+                    option_to_data[curr_option.name].append(
+                        (curr_states, curr_actions, curr_rewards,
+                         curr_relative_params))
+
+                    # Advance to next option.
+                    curr_option_idx += 1
+                    if curr_option_idx < len(plan):
```

Comment:
I'm hoping that in addition to simplifying these lines with the refactoring suggested above, we can do additional refactoring to remove any duplicated logic (this all looks similar to code above)